src/common/snippets/include/snippets/op/loop.hpp

```diff
@@ -0,0 +1,97 @@
+    // if set:
+    // * LoopEnd doesn't pop reg_work_amount from the stack on exit (to be disabled after assign_registers refactoring)
+    // * LoopBegin doesn't set reg_work_amount (it assumes it's already correct)
+    bool reuse_work_amount_reg;
+
+protected:
+    size_t dimension;
+    size_t work_amount;
+    size_t increment;
+    bool evaluate_once; // true if the Loop is executed only once, used to skip setting and testing the loop counter
+};
+class LoopEnd;
+class LoopBegin : public LoopBase {
+    friend LoopEnd;
+public:
+    OPENVINO_OP("LoopBegin", "SnippetsOpset");
+    /// \brief Construct an Loop
+    /// \param region The vector of pairs: emitters and the corresponding registers
+    /// \param increment Loop size - count of elements to load and store.
+    ///                  Vector Loop should have size of vector register and Scalar Loop should have 1
+    /// \param num_inputs Count of inputs
+    /// \param num_outputs Count of outputs
+    /// \param io_dims Vector of last dimensions of inputs and outputs
+    /// \param io_data_sizes Vector of data type sizes of inputs and outputs
+    explicit LoopBegin(const std::vector<Output<Node>>& args);
+    LoopBegin() = delete;
+    void validate_and_infer_types() override;
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& inputs)  const override;
+    std::shared_ptr<LoopEnd> get_loop_end();
+    // begin_address and input_regs are needed to communicate information between LoopBegin and LoopEnd emitters
+    const uint8_t* begin_address;
+    std::vector<size_t> input_regs;
+private:
+    void validate_and_infer_types_except_LoopEnd();
+    LoopBegin(const std::vector<Output<Node>>& args, size_t dimension, size_t work_amount, size_t increment);
+};
+
+class LoopEnd : public LoopBase {
+public:
+    OPENVINO_OP("LoopEnd", "SnippetsOpset");
+    LoopEnd(const std::vector<Output<Node>>& args, size_t dimension, size_t work_amount, size_t increment,
+              std::vector<bool> apply_increment, std::vector<int64_t> finalization_offsets);
+    LoopEnd() = delete;
+    std::shared_ptr<LoopBegin> get_loop_begin();
+    void validate_and_infer_types() override;
+    std::shared_ptr<Node> clone_with_new_inputs(const OutputVector& inputs)  const override;
+    const std::vector<int64_t>& get_finalization_offsets() const;
+    const std::vector<bool>& get_apply_increment() const;
+    void set_finalization_offsets(std::vector<int64_t> offsets);
+    void set_apply_increment(std::vector<bool> apply_increment);
+    void set_work_amount(size_t new_work_amount);
+    void set_increment(size_t new_increment);
+    void set_evaluate_once(bool once);
+    // Used to propagate information about Loop structure, needed to simplify some optimizations. For example,
+    // to skip pointer increments when outer Loop is empty, and work_amount == vector_size (one inner vector Loop)
+    // true by default, the optimizations enabled if it's false;
+    bool has_outer_loop;
```

Comment:
As I understand this parameter is used for optimizations and setted only in `InsertLoop` pass. If I want to explicitly insert Loop (for example, for Softmax) I should know and remember about this parameter.
But unfortunately this remark is missed. Could you please add comment about it? That we should manually set value. Or maybe we can calculate this parameter inside ctor. What do you think?
Correct me please if I'm wrong

Reply:
There is a comment right where the field is declared: https://github.com/IvanNovoselov/openvino/blob/fa9660b02c522ca2d13483aeb20c753f7016aab1/src/common/snippets/include/snippets/op/loop.hpp#L84-L87
Please specify where do you want to see the comment?

Reply:
Regarding automatic calculation, I don't think it's a right thing to do. Setting this flag to non-default value (false) enables some optimizations (allows to skip pointer increments), but these optimizations imply that no one would need to access data stored by gprs. That is often true, but not always (e.g. you can have other ops after Tile). Therefore automatic enabling of these optimizations might trigger some hard-to-trace segfaults in future.